### ISSUE TITLE
feat(ledger-connector-corda): support Corda RPC natively

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cactus#280]
Depends on #143 

**Is your feature request related to a problem? Please describe.**

Right now the corda connector uses SSH and CLI tools to deploy contracts and run flows while all of these should be doable through the RPC interface in theory.
Problem is, there is no Javascript bindings written by the maintainers of Corda.

**Describe the solution you'd like**

Would be nice to have our own bindings compiled from Corda's Kotlin code straight to Javascript and then we could use that instead of hacky solutions like graalvm or a separate JVM process, both of which stinks because of the additional sidecar-ish OS process that it requires.

I started out with a corda fork where I attempted to add the kotlin-js gradle plugin to compile their RPC package, but it didn't go well. More details here (for now): https://stackoverflow.com/questions/63574532/gradle-kotlin-js-plugin-could-not-generate-decorated-class

**Describe alternatives you've considered**

GraalVM and JVM processes, similar how we had it in the original codebase that came out of Accenture labs.

**Additional context**

Currently the PR is in progress on which this issue itself depends on (e.g. this issue doesn't really exist in the upstream until we merge #185 but once that happens it is on)

Relevant RocketChat exchange excerpt:

```sh
sfuji822
6:16 PM
jweate peter_somogyvari I see PR#190 https://github.com/hyperledger/cactus/pull/190 is still 
have a issue to be merged. I think you need to execute rebase for passing commit check. 
Since we would like to use your plugin at the demo application, please tell us about your 
plan to fix the issue.

peter_somogyvari
10:01 AM
sfuji822 That one is still in progress unfortunately (it is just a draft, so not ready for review).
My latest info is that the container image itself is operational, but the test automation is 
still being finalized.

Furthermore there's this technical challenge that is unsolved by this PR about the RPC communication 
between Cactus and the Corda nodes/apps. There are basically 3 options to communicate from the 
NodeJS code with Corda

1) host a Java app as an in-between translation layer (because Corda RPC libraries are written in Kotlin)

2) run the java app in graalvm and directly invoke the RPC related code from NodeJS

3) compile the RPC corda code from Kotlin to JS and use that as a NodeJS library directly in Cactus.

4) use SSH to connect to the corda server and execute tasks via the CLI (such as 
deploying contracts, starting flows etc)

In the #190 PR we went with option 4) because it was the simplest for a proof of concept, but longer 
term of course this will not scale well for high throughput, performance sensitive production 
deployments so my plan is to work on option 3) as the most maintainable option.

Option 1 and 2 both require external processes to be running in tandem to the Cactus process 
so both of those are sub-optimal when it comes to maintenance and operational complexity 
because of the extra processes (also bad for containerization because by design a container's 
life-cycle is tied to a single process running in the container)

That's the context so far. What I could use help with is the implementation of option 3) and if 
that turns out to be very high effort then option 2 as a fallback/last resort.

Regarding my try on option 3 so far: Tried to add the kotlin JS plugin to my corda fork, but the 
build then crashes with some gradle dependency error that I haven't yet had the time to investigate properly.
```

cc: @sfuji822 @takeutak @jonathan-m-hamilton

**Comment by petermetz** [comment]
Update: Spent a lot of time working on this recently and it's been a long journey.
What I settled on is braid, but it's not fully operational at the moment so we'll have to wait a little longer on that.

**Comment by kikoncuo** [comment]
Looking for new ways to handle corda interaction using jar binaries until next corda version offers native support for this.
